builder/frameworks/arduino.py

```
@@ -369,6 +369,23 @@ def get_MD5_hash(phrase):
+    sdkconfig but are not part of custom_sdkconfig. They must affect the checksum
+    too, otherwise a PSRAM board and a no-PSRAM board sharing the same
+    custom_sdkconfig would reuse each other's precompiled Arduino libs."""
+    build_section = board.get("build", {})
+    memory_type = (build_section.get("arduino", {}).get("memory_type", "")
+                   or build_section.get("memory_type", ""))
+    psram_type = build_section.get("psram_type", "")
+    try:
+        memory_type = env.GetProjectOption("board_build.memory_type", "") or memory_type
+        psram_type = env.GetProjectOption("board_build.psram_type", "") or psram_type
+    except Exception:
+        pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)